scripts/evaluate_string_segments.py

```diff
@@ -0,0 +1,321 @@
+        '-b',
+        '--build-path',
+        dest='build_path',
+        type=str,
+        required=True,
+        help='Path where the executables to benchmark are located.'
+    )
+    parser.add_argument(
+        '-s',
+        '--scale-factor',
+        dest='scale_factor',
+        type=float,
+        required=True,
+        help='The scale factor to pass to the benchmarks that support scaling. Note that this number might get rounded or ignored if necessary for a benchmark.'
+    )
+    parser.add_argument(
+        '-p',
+        '--tmp-path',
+        dest='tmp_path',
+        type=str,
+        required=False,
+        default='tmp',
+        help='The directory where the benchmark result files will be stored.'
+    )
+    parser.add_argument(
+        '-e',
+        '--encoding-under-test',
+        dest='encoding',
+        type=str,
+        required=True,
+        help='The name of the encoding to compare against built-in encodings.'
+    )
+    parser.add_argument(
+        '-t',
+        '--timeout',
+        dest='timeout',
+        type=int,
+        required=False,
+        default=-1,
+        help='The timeout in seconds to pass to the benchmarks. Defaults to -1, i.e. no timeout.'
+    )
+    parser.add_argument(
+        '-d',
+        '--debug',
+        dest='debug',
+        required=False,
+        default=False,
+        action=BooleanOptionalAction,
+        help='Whether to activate debug mode (more verbose output to stdout).'
+    )
+    namespace = parser.parse_args()
+    if namespace.debug:
+        DEBUG_MODE = True
+        print_debug('Debug Mode enabled.')
+    return Configuration(
+        output_file=namespace.output_file,
+        build_path=namespace.build_path,
+        tmp_path=namespace.tmp_path,
+        scale_factor=namespace.scale_factor,
+        encoding_under_test=namespace.encoding,
+        time_limit=namespace.timeout,
+        debug=namespace.debug)
+
+
+def scale_factor_for_benchmark(benchmark: str, scale_factor: float) -> float:
+    if benchmark == 'hyriseBenchmarkTPCDS':
+        return max(scale_factor, 1)
+    return scale_factor
+
+
+def locate_benchmarks(benchmarks: list[str], config: Configuration) -> list[Benchmark]:
+    benchmark_objects: list[Benchmark] = []
+    for benchmark in benchmarks:
+        benchmark_path = path.join(config.build_path, benchmark)
+        if not path.isfile(benchmark_path):
+            exit(f'Cannot locate {benchmark} at {benchmark_path}!')
+        benchmark_objects.append(Benchmark.create(benchmark))
+    return benchmark_objects
+
+
+def main():
+    global output_file
+    config = parse_arguments()
+    with open(config.output_file, 'w+') as output_file:
+        benchmarks_names = ['hyriseBenchmarkTPCH', 'hyriseBenchmarkTPCDS', 'hyriseBenchmarkJoinOrder', 'hyriseBenchmarkStarSchema']
+        benchmarks = locate_benchmarks(benchmarks_names, config)
+        for benchmark in benchmarks:
+            benchmark.run(config)
```

Comment:
There is `-d` that will print much more information, however, it will also print the output of the commands. I'll look into splitting those information into two verbosity levels.